docs/contributor_guide/contributing_example_notebooks.rst

```diff
@@ -17,16 +17,60 @@ A good example notebook exhibits the following attributes:
    Avoids abstraction traps.
    See `Selbst et al. (2020) <https://andrewselbst.files.wordpress.com/2019/10/selbst-et-al-fairness-and-abstraction-in-sociotechnical-systems.pdf>`_.
 4. **Substantiated**: Discusses trade-offs and compares alternatives.
-   Describes why using particular Fairlearn functionalities makes sense.
+   Describes why using particular Fairlearn functionalities makes sense if
+   Fairlearn is used.
 5. **For developers**: Speaks the language of developers and data scientists.
    Considers real practitioner needs.
-   Fits within the lifecycle of real practioner work.
+   Fits within the lifecycle of real practitioner work.
```

Comment:
examples are sometimes there just to show how somebody could use the code, they don't have to be a real usecase. Sometimes showing how one can use a part of the API doesn't need to show why one needs to use that part of the API.

Reply:
I guess that's that leads to the bigger question of whether it's responsible to show someone how to use something without explaining when/how and when/how not to use it at the same time. If something is really just about "how one can use a part of the API" that's what the user guide (or perhaps even API doc) is for, isn't it?

Reply:
To me the user guides can go more into detail than examples. Examples are mostly about the code, whereas the user guide gives context and talks about stuff.

Reply:
That's not really how we've been talking about examples as you can tell from that entire file. I'm not really writing this from scratch here either, it's been there for a while and I'm just making updates. The way we've talked about examples for Fairlearn as a "special" kind of open source project with a sociotechnical focus is that examples are mainly for showing practitioners how to use this properly/responsibly rather than "just code".

Let's loop in @mmadaio @MiroDudik @riedgar-ms @hildeweerts since this is a rather big question. Wouldn't want that to get lost.

Reply:
Yeah... We had a long conversation about this a while back. At the time we concluded that example notebooks should be more like use cases rather than short code samples. We could revisit this question. But rather than discussing these requirements hypothetically, I'd be in favor of doing it in context of a concrete example that would fall into such a "short sample code" category, but fail our current notebook requirements?

My current thinking around where different kinds of examples fit into our documentation:
1. **API docs** [.py]: The docs describe technical functionality in isolation, and so I think that they are the best place for short examples of functionality. However, they might not be the right place for examples that deal with several parts of Fairlearn (e.g., compare several algorithms).
2. **User guide** [.rst]: I think about this basically as a tutorial, almost a textbook. Here it should be pretty straightforward to provide context and also have sections that can be more technically focused. Unlike API docs, this might be a good place to present examples that mix different types of functionality.
3. **Example notebooks** [.ipynb] (note that we haven't been calling these "Examples"): They are basically use cases and what we're currently describing in [Contributing example notebooks](https://fairlearn.github.io/v0.5.0/contributor_guide/contributing_example_notebooks.html).

Unfortunately, `.ipynb` is not GitHub friendly, and so example notebooks are `.py`.

Reply:
I like @MiroDudik's thinking. I'd suggest to be very explicit in naming and call examples in (1) "code examples" and also add a link to the relevant section in the user guide (this would only work if we're giving every object/function, but I think that's something we want to work on anyways, right?).

Reply:
Yes! I'll rewrite this PR based on today's discussion in the community call :-) Stay tuned. I'll make sure to ping you all when I've worked in the changes.